### PR TITLE
Remove the 'zowe profiles' command command handlers & APIs

### PR DESCRIPTION
**Pull request by gejohnston**
**What It Does**

This PR removes the command handlers for the `zowe profiles` command group.

Also:
- Numerous APIs, interfaces, and constants that supported V1 profiles were also removed.

- Three classes (AbstractProfileManager which was extended by BasicProfileManager, which was extended by CliProfileManager) were consolidated into the one CliProfileManager class.

- The `--show-inputs-only` options and the `config report-env` commands no longer report V1 profile information.

- Replaced the term "Team configuration" with "Zowe client configuration" in the `zowe config report-env` command.

- Changed a reference in the `config convert-profiles` command  from "V2 profiles" to "TeamConfig".

- Removed the display of the NVM version number in the `config report-env` command. Newer versions of NVM display a GUI popup message when launched as a subprocess, which complains that NVM should be run from a terminal (version 1.1.12 displays this behavior). The popup blocks further output until the user responds to the popup. NVM is *NOT* used by customers frequently enough to warrant an exhaustive effort to keep that piece of information in the report. For awareness, the NVM dialog is shown below.

    image

**How to Test**

1. Commands should continue to run as before when no configuration exists or a team configuration exists. Commands should ignore V1 profiles, even when they are the only available configuration. In particular, if only V1 profiles exist, running a Zowe command now will prompt the user for connection information.
    ```
    > zowe files list data-set "SYS1.PARMLIB*"
    Enter the host name of your service: YourHostName
    Enter the user name for your service (will be hidden):
    Enter the password for your service (will be hidden):
    ```
2. Use `--show-inputs-only`, `config report-env`, & `config convert-profiles` and observe that no references to V1 or V2 profiles are displayed.

#### Follow-ups

- The "ProfileIO" class must be renamed to "V1ProfileConversion", and all write functions must be removed.
- Any additional V1 logic that is no longer called should be removed.

**Review Checklist**
I certify that I have:
- [x] tested my changes
- [x] added/updated automated tests
- [x] updated the changelog
- [x] followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)

**Additional Comments**

### CODE REVIEW

File: packages/imperative/__tests__/src/packages/cmd/__integration__/CliProfileManager.integration.test.ts [hunk 1]
```diff
@@ -9,47 +9,67 @@
...
+        }
+
+        expect(newProfMgr).not.toBeNull();
+        expect(caughtError.message).toEqual("");
+    });
+
+    it("should be able to retrieve all defined types after init", async function () {
+        const Imperative = require("../../../../../src/imperative/src/Imperative").Imperative;
+        const ImperativeConfig = require("../../../../../src/utilities/src/ImperativeConfig").ImperativeConfig;
+        const ImperativeError = require("../../../../../src/error/src/ImperativeError").ImperativeError;
```
[the remark this answers is not in the document]
Comment (awharn) [reply packages/imperative/__tests__/src/packages/cmd/__integration__/CliProfileManager.integration.test.ts:67]: CodeQL says ImperativeError is not used.

File: packages/imperative/src/profiles/src/utils/ProfileUtils.ts [hunk 2]
```diff
@@ -18,6 +18,7 @@ import * as nodePath from "path";
  * Set of static utility methods to assist with creating profile option names from profile types, constructing the
  * root directory, reforming responses for different purposes, etc.
  *
+ * @internal
```
Comment (awharn) [inline packages/imperative/src/profiles/src/utils/ProfileUtils.ts:21]: I know we recently had a change that something like this broke Zowe Explorer. Do we see this having the same effect? Is there something Imperative exports for full conversion?

File: packages/imperative/src/profiles/src/validation/api/ProfileValidator.ts [hunk 3]
```diff
@@ -28,6 +28,7 @@ import { CliUtils } from "../../../../utilities/src/CliUtils";
 /**
  * API for going through the full validation test for a Zowe CLI profile
  * and producing validation report
+ * @internal
```
Comment (awharn) [inline packages/imperative/src/profiles/src/validation/api/ProfileValidator.ts:31]: Same question as above

File: packages/imperative/src/profiles/src/utils/ProfileUtils.ts (hunk 2 above)
Comment (gejohnston) [reply packages/imperative/src/profiles/src/utils/ProfileUtils.ts:21]: A quick search of ZE code shows no import of ProfileUtils. ZE should use imperative's ConfigBuilder.convert() to convert V1 profiles. That function does not rely on ProfileUtils.

File: packages/core/src/utils/ProfileUtils.ts [hunk 4]
```diff
@@ -1,109 +0,0 @@
...
-    const defaultHome = path.join(os.homedir(), ".zowe");
-    if (ImperativeConfig.instance.loadedConfig?.defaultHome !== defaultHome) {
-        ImperativeConfig.instance.loadedConfig = {
-            name: "zowe",
-            defaultHome,
-            envVariablePrefix: "ZOWE"
-        };
-    }
-    return ImperativeConfig.instance.cliHome;
-}
```
Comment (zFernand0) [inline packages/core/src/utils/ProfileUtils.ts:31]: Curious if we need to create an equivalent `getZoweDir` method in the ProfileInfo APIs for Zowe Explorer

Please note that there is only one reference in Zowe Explorer to this method
- https://github.com/zowe/vscode-extension-for-zowe/blob/next/packages/zowe-explorer-api/src/utils/FileManagement.ts#L36-L38

```
    public static getZoweDir(): string {
        return zowe.getZoweDir();
    }
```

So we could just move it over there 😋

File: packages/imperative/src/config/src/ProfileInfo.ts [hunk 5]
```diff
@@ -959,58 +823,10 @@ export class ProfileInfo {
...
-     * old-school profiles.
-     *
-     * You must call ProfileInfo.readProfilesFromDisk() before calling this function.
-     *
-     * @returns True when we are using a team config. False means old-school profiles.
-     */
-    public get usingTeamConfig(): boolean {
-        this.ensureReadFromDisk();
-        return this.mUsingTeamConfig;
-    }
```
Comment (zFernand0) [inline packages/imperative/src/config/src/ProfileInfo.ts:1012]: There were also a few references to the `ProfileInfo.usingTeamConfig`

- https://github.com/zowe/vscode-extension-for-zowe/blob/next/packages/zowe-explorer-api/src/profiles/ProfilesCache.ts#L153-L155
- `packages/zowe-explorer/src/utils/ProfilesUtils.ts(418,26)`
- `packages/zowe-explorer/src/utils/ProfilesUtils.ts(475,26)`
- `packages/zowe-explorer/src/utils/ProfileManagement.ts(292,22)`
- `packages/zowe-explorer/src/Profiles.ts(889,22)`

Since this is vNext, it doesn't make much sense to keep this reference.
Thus I suggest using the new `ProfileInfo.onlyV1ProfilesExist` to prevent V1 operations going forward.

**Review by zFernand0 — APPROVED** [review]
TL;DR: LGTM! 😋 

---

I've been lucky enough to have had tested this PR thanks to the other PRs (#2067 and a little bit in #2062) that depended on this branch (`rm-profile-handlers`)

I might continue checking other parts of this PR tomorrow 😋 

In ZE you can find a branch called `prep/rm-profile-handlers`. This should contain a "working" version of the extension with this branch linked up.

---

I ran into a runtime problem after loading ZE, however I do believe it is due to how I linked the project. For exmaple, I just had a bunch of PNPM package overrides that looked kind of like this:
- "@zowe/core-for-zowe-sdk": "file:///root/gh/zowe/zowe-cli/packages/core",

In case it rings any bells...
TL;DR: `TypeError: Cannot read properties of undefined (reading 'AsymmetricMatcher')`

Here is a partial stack trace of the error.
```
TypeError: Cannot read properties of undefined (reading 'AsymmetricMatcher')
at ../../node_modules/.pnpm/jest-diff@24.9.0/node_modules/jest-diff/build/index.js (/root/gh/zowe/vscode-extension-for-zowe/packages/zowe-explorer/out/src/vendors.extension.js:797:1904)
at r (/root/gh/zowe/vscode-extension-for-zowe/packages/zowe-explorer/out/src/runtime.extension.js:12:144)
at ../../node_modules/.pnpm/file+..+zowe-cli+packages+imperative/node_modules/@zowe/imperative/lib/utilities/src/diff/DiffUtils.js (/root/gh/zowe/vscode-extension-for-zowe/packages/zowe-explorer/out/src/vendors.extension.js:597:2033)
at r (/root/gh/zowe/vscode-extension-for-zowe/packages/zowe-explorer/out/src/runtime.extension.js:12:144)
at ../../node_modules/.pnpm/file+..+zowe-cli+packages+imperative/node_modules/@zowe/imperative/lib/utilities/index.js (/root/gh/zowe/vscode-extension-for-zowe/packages/zowe-explorer/out/src/vendors.extension.js:584:9595)
at r (/root/gh/zowe/vscode-extension-for-zowe/packages/zowe-explorer/out/src/runtime.extension.js:12:144)
at ../../node_modules/.pnpm/file+..+zowe-cli+packages+imperative/node_modules/@zowe/imperative/lib/cmd/src/help/abstract/AbstractHelpGenerator.js (/root/gh/zowe/vscode-extension-for-zowe/packages/zowe-explorer/out/src/vendors.extension.js:332:2099)
at r (/root/gh/zowe/vscode-extension-for-zowe/packages/zowe-explorer/out/src/runtime.extension.js:12:144)
at ../../node_modules/.pnpm/file+..+zowe-cli+packages+imperative/node_modules/@zowe/imperative/lib/cmd/index.js (/root/gh/zowe/vscode-extension-for-zowe/packages/zowe-explorer/out/src/vendors.extension.js:195:6147)
at r (/root/gh/zowe/vscode-extension-for-zowe/packages/zowe-explorer/out/src/runtime.extension.js:12:144)
```

If you look closer, you might see a bunch of `index.js` references from the `file:///root/gh/...../imperative` package

File: packages/core/src/utils/ProfileUtils.ts (hunk 4 above)
Comment (gejohnston) [reply packages/core/src/utils/ProfileUtils.ts:31]: I removed ProfileUtils.ts because I found no use of any of its functions (including getZoweDir). I missed ZE's use of getZoweDir. I could resurrect ProfileUtils.ts with just getZoweDir, or place getZoweDir into profileUtils.ts as you suggested. Do you think one choice is better than the other?

Either way, I think one of those changes should be made before merging this PR into next. Thanks.

File: packages/imperative/src/config/src/ProfileInfo.ts (hunk 5 above)
Comment (gejohnston) [reply packages/imperative/src/config/src/ProfileInfo.ts:1012]: Even though this requires changes for consumers, I agree that we should all completely stop using ProfileInfo.usingTeamConfig. It is confusing to someone reading the logic. Does it mean we should be trying to use a team config, or does it mean that we have actually found an existing team config? To reinforce this, the following text is part of our breaking changes document:

Removed:
- ProfileInfo.usingTeamConfig
    - To detect if a team config exists use ProfileInfo.getTeamConfig
    - To detect if only V1 profiles exist use ProfileInfo.onlyV1ProfilesExist

Finally, it is possible that some of those ZE references should be just be removed. In many cases ZE should only be trying to read a team config, so there may be no need for a test.

**Review by traeok — APPROVED** [review]
LGTM - thanks @gejohnston! 😁 

I had a couple of questions regarding an integration test case and the type definition for one of the profile-related properties, but nothing that should hold up this PR.

File: packages/imperative/__tests__/src/packages/profiles/__integration__/ProfileCommandExample.integration.test.ts [hunk 6]
```diff
@@ -129,7 +127,7 @@ describe("Imperative should allow CLI implementations to configure their own pro
     }
 
     // eslint-disable-next-line jest/expect-expect, jest/no-disabled-tests
-    it.skip("We should be able to issue a command and have a profile be automatically loaded", function () {
+    it("We should be able to issue a command and have a profile be automatically loaded", function () {
         // return issueCommand.bind(this, ["profile"], true)();
     });
```
[the remark this answers is not in the document]
Comment (gejohnston) [reply packages/imperative/__tests__/src/packages/profiles/__integration__/ProfileCommandExample.integration.test.ts:132]: My only changes were to make this file compile after some things were removed. So I never paid much attention to the rest of the content of this file. It appears to have been in this state since well before I started any of my current work. Since this test, which does nothing, is the only test in this file, I think the right action is to remove the entire file. That is what I will do. Thanks for noticing the weirdness.

File: packages/imperative/src/cmd/src/CommandProcessor.ts [hunk 7]
```diff
@@ -64,10 +63,10 @@ interface IResolvedArgsResponse {
 
     /**
      * Whether we're using old profiles or config
-     * @type {(`v1` | `v2`)}
+     * @type {(`v1` | `TeamConfig`)}
      * @memberof IResolvedArgsResponse
      */
-    profileVersion?: `v1` | `v2`;
+    profileVersion?: `v1` | `TeamConfig`;
```
[the remark this answers is not in the document]
Comment (gejohnston) [reply packages/imperative/src/cmd/src/CommandProcessor.ts:69]: I changed this item as part of a search & replace of 'V2' to 'TeamConfig'. Your comment prompted me to search for all uses of 'profileVersion'. There are none. I will remove 'profileVersion' property.

File: packages/imperative/CHANGELOG.md [hunk 8]
```diff
@@ -2,6 +2,188 @@
...
+      - IUpdateProfile
+      - IUpdateProfileFromCliArgs
+      - IValidateProfile
+      - IValidateProfileForCLI
+      - IValidateProfileWithSchema
+
+  - Removed the following obsolete V1 profile classes/functions:
+    - @zowe/core-for-zowe-sdk
+      - getDefaultProfile
+      - getZoweDir
```
[the remark this answers is not in the document]
Comment (gejohnston) [reply packages/imperative/CHANGELOG.md]: Agreed. It will be a static method.

**Review by awharn — APPROVED** [review]

File: packages/imperative/src/cmd/src/profiles/CommandProfiles.ts [hunk 9]
```diff
@@ -15,8 +15,7 @@ import { ImperativeExpect } from "../../../expect";
 
 /**
  * Profiles map created by the command profile loader and passed to the handler via parameters. Handlers can
- * retreive loaded profiles from the map via the profile type.
- * @export
+ * retrieve loaded profiles from the map via the profile type.
  * @class CommandProfiles
  */
 export class CommandProfiles {
```
Comment (t1m0thyj) [inline packages/imperative/src/cmd/src/profiles/CommandProfiles.ts:21]: Are the `CommandProfiles` and `CommandProfileLoader` classes still needed? I thought they only existed for the purpose of working with v1 profiles.

**Review by t1m0thyj — APPROVED** [review]

File: packages/imperative/src/profiles/src/utils/ProfileIO.ts [hunk 10]
```diff
@@ -312,7 +312,7 @@ export class ProfileIO {
     private static crashInTeamConfigMode() {
         if (ImperativeConfig.instance.config?.exists) {
             try {
-                throw new Error("A Zowe V1 profile operation was attempted with a Zowe V2 configuration in use.");
+                throw new Error("A Zowe V1 profile operation was attempted with a Zowe team configuration in use.");
```
[the remark this answers is not in the document]
Comment (gejohnston) [reply packages/imperative/src/profiles/src/utils/ProfileIO.ts:315]: There is no official policy from Rose, Adam, and Ana on the terminology that we should use for our configuration. 
In V2, we needed to differentiate between our old and new configuration, the terms "V1 profiles" and "Team Configuration" made sense. Since we should be removing our references to V1 profiles, I decided to use our command output (whose text is not the primary focus of policy makers) to refer to our one and only available configuration as the "Zowe client configuration". I made an exception in the convert-profiles command to differentiate that we are converting V1 profiles to something distinctively different. Thus I kept the term "Team Configuration" in some output of that particular command. If the majority of people feel that my choice of terminology is not desired, we could revise the command output across the board in our next story.

The particular case that you sited was in ProfileIO.ts. That file was not overhauled in this PR. It will be the primary focus of the next story on V1 profiles. It is possible that we may remove that exception, maybe the entire function (I do not know for sure yet). However we resolve it, this specific message should be addressed during the next V1 story.

File: packages/imperative/src/cmd/src/profiles/CommandProfiles.ts (hunk 9 above)
Comment (gejohnston) [reply packages/imperative/src/cmd/src/profiles/CommandProfiles.ts:21]: It might be possible to remove them. However, CommandProfileLoader is called by the CommandProcessor. CommandProfileLoader implies that it gathers profile information from the command's definition, which might still be relevant. I was afraid to remove it. I did mark CommandProfileLoader as @internal making it possible for us to remove it in the future without breaking consumers.

CommandProfileLoader references CommandProfiles. CommandProfiles are also referenced by numerous interfaces. Some of those interfaces are outside of the imperative package, so I could not mark CommandProfiles as @internal.

I think that the removal of these two items may require a lot change and a lot time to correct. I was not convinced that the extra baggage was worth the extra time needed to remove it. In our next V1 profiles story, after completing its primary objectives, we could decide as a team if we want to take some extra time to try to remove these items during that story. I will make a note to revisit this subject at that time.
